trunk/examples/CToBoogieTranslation/regression/all/varargsDifferentTypes2.c

```diff
@@ -0,0 +1,24 @@
+//#Safe
+/*
+  Author: Frank Schüssele (schuessf@informatik.uni-freiburg.de)
+  Date: 2022-12-07
+*/
+
+void check(int count,...) {
+   int result = 0;
+   
+   __builtin_va_list valist;
+   __builtin_va_start(valist, count);
+   
+   char* string = __builtin_va_arg(valist, char*);
+   int actuallyChar = __builtin_va_arg(valist, int);

```

Comment:
Is this defined behaviour?

Reply:
You mean "casting" the argument to `int` that has actually type `char` in the call? Yes, since all types smaller than int are promoted to int.